Revamp JSON BBEs (#3878)
ballerina-platform/ballerina-distribution: examples/resource-method-typing/resource_method_typing.md

# Resource method typing

> Resource method arguments can use user-defined types. Listener will use introspection to map from protocol format (typically JSON) to user-defined type, using cloneWithType. Return value that is a subtype of anydata will be mapped from user-defined type to protocol format, typically JSON, using toJson.

Reviewer: ```suggestion
Resource method arguments can use user-defined types. The listener will use introspection to map the protocol format (typically JSON) to a user-defined type using `cloneWithType`. The return value, which is a subtype of anydata will be mapped from the user-defined type to the protocol format typically JSON, using `toJson`.
```

> Can generate API description (e.g. OpenAPI) from Ballerina service declaration. Annotations can be used to refine the mapping between Ballerina-declared type and wire format.

Reviewer: ```suggestion
The API description (e.g. OpenAPI) can be generated from the Ballerina service declaration. Annotations can be used to refine the mapping between the Ballerina-declared type and wire format.
```

## Related links

> - [Converting from JSON to user-defined type](/learn/by-example/converting-from-json-to-user-defined-type/)

Reviewer 2: ```suggestion
- [Casting JSON to user-defined type](/learn/by-example/casting-json-to-user-defined-type/)
```
